### DISCUSSION TITLE
text becomes unselectable from selectable

### DISCUSSION THREAD

**Discussion by RottenYogurt** [sumatrapdfreader/sumatrapdf#3995] [Q&A]
I have a pdf that I've been reading and all the text can be selected including odd looking text which is a cod example. But after I changed the name of this pdf file, that all odd looking code example becomes unselectable, I think those become images. I'm so embarrassed by this sudden changes. Is this any known issue? Should I execute ocr at the all images?

**Comment by GitHubRulesOK** [comment]
simply changing a filename should not affect the PDF internal behaviour unless using a method that included a file rewrite?

**Reply by RottenYogurt** [reply]
that's why I'm so embarrassed. Now some of the codes cannot be identified that is split by page break. For real, I just opened the file right after changing file name, it's all I did. No action can be happen in the meantime.

**Reply by GitHubRulesOK** [reply]
without a copy of the file to look at its hard to guess what may be a problem.

**Reply by RottenYogurt** [reply]
I'm worried if there's no problem uploading this file, this pdf file is a book: Java_ A Beginner’s Guide, Eighth Edition - Schildt, H.-McGraw-Hill Education (2018).

**Reply by GitHubRulesOK** [reply]
The copy I download shows the coded areas like this are images not text.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/888451e4-0318-4ca0-bc77-af8784ac2e91)

this appears to be common with these file types, sometimes there is a companion cd or download for the coded contents. 

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/51c8815b-acfe-452d-a5f5-2e7ce47f8348)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/4259ede2-82cb-4fc7-aab3-4fc829d8963b)

**Reply by RottenYogurt** [reply]
Right, but it is so wierd until yesterday, I've copied all the code and tested up to 'Generic' section. I know it's hard to believe for person who downloaded that file and checked it's code example contents are all images, then what are the example codes that I tested and saved in my disk? Even I'm confused like "wasn't that file what I was reading?". I can't believe this situation.

**Reply by GitHubRulesOK** [reply]
The coded sections are often not included so my copy states

>Screen displays of copyrighted Oracle software programs have been reproduced herein with the permission of Oracle Corporation and/or its affiliates.

However it is possible you have /had some screen copy app active MS have one included in their PowerToys.
